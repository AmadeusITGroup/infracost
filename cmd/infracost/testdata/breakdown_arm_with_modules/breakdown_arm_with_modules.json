{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "metadata": {
    "_generator": {
      "name": "example",
      "version": "0.27.1.19265",
      "templateHash": "4270386830956032562"
    }
  },
  "parameters": {
      "rgLocation": {
        "type": "string"
      },
      "vmName": {
        "type": "string"
      },
      "osDisk": {
        "type": "object"
      }
  },
  "resources": [
    {
      "type": "Microsoft.Resources/deployments",
      "apiVersion": "2022-09-01",
      "name": "module",
      "location": "eastus",
      "properties": {
          "expressionEvaluationOptions": {
            "scope": "inner"
          },
          "mode": "Incremental",
          "parameters": {
            "rgLocation": {
              "value": "[parameters('rgLocation')]"
            },
            "vmName": {
              "value": "[parameters('vmName')]"
            },
            "osDisk": {
              "value": "[parameters('osDisk')]"
            }
          },
          "template": {
            "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
            "contentVersion": "1.0.0.0",
            "metadata": {
              "_generator": {
                "name": "bicep",
                "version": "0.26.54.24096",
                "templateHash": "12573701218424896755"
              }
            },
            "parameters": {
              "rgLocation": {
                "type": "string"
              },
              "vmName": {
                "type": "string"
              },
              "osDisk": {
                "type": "object"
              }
            },
            "resources": [
              {
                  "type": "Microsoft.Compute/virtualMachines",
                  "apiVersion": "2023-09-01",
                  "name": "[parameters('vmName')]",
                  "location": "[parameters('rgLocation')]",
                  "properties": {
                    "hardwareProfile": {
                      "vmSize": "Basic_A2"
                    },
                    "storageProfile": {
                      "imageReference": {
                        "publisher": "Canonical",
                        "offer": "UbuntuServer",
                        "sku": "16.04-LTSr",
                        "version": "latest"
                      },
                      "osDisk": "[parameters('osDisk')]"
                    },
                    "osProfile": {
                          "computerName": "basic_b1",
                          "adminUsername": "fakeuser",
                          "adminPassword": "Password1234!"
                      },
                    "networkProfile": {
                      "networkInterfaces": [
                        {
                          "id": "[resourceId('Microsoft.Network/networkInterfaces', '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testrg/providers/Microsoft.Network/networkInterfaces/fakenic')]"
                        }
                      ]
                    }
                  }
              },
              {
                "type": "Microsoft.Resources/deployments",
                "apiVersion": "2022-09-01",
                "name": "secondmodule",
                "properties": {
                  "expressionEvaluationOptions": {
                    "scope": "inner"
                  },
                  "mode": "Incremental",
                  "parameters": {
                    "rgLocation": {
                      "value": "[parameters('rgLocation')]"
                    }
                  },
                  "template": {
                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
                    "contentVersion": "1.0.0.0",
                    "metadata": {
                      "_generator": {
                        "name": "bicep",
                        "version": "0.26.54.24096",
                        "templateHash": "5635318929995614965"
                      }
                    },
                    "parameters": {
                      "rgLocation": {
                        "type": "string"
                      }
                    },
                    "resources": [
                      {
                          "type": "Microsoft.Compute/virtualMachines",
                          "apiVersion": "2023-09-01",
                          "name": "basic_a2",
                          "location": "[parameters('rgLocation')]",
                          "properties": {
                            "hardwareProfile": {
                              "vmSize": "Basic_A2"
                            },
                            "storageProfile": {
                              "imageReference": {
                                "publisher": "Canonical",
                                "offer": "UbuntuServer",
                                "sku": "16.04-LTSr",
                                "version": "latest"
                              },
                              "osDisk": {
                                "createOption": "FromImage",
                                "managedDisk": {
                                  "storageAccountType": "Standard_LRS",
                                  "caching": "ReadWrite"
                                }
                              }
                            },
                            "osProfile": {
                              "computerName": "basic_a2",
                              "adminUsername": "fakeuser",
                              "adminPassword": "Password1234!"
                            },
                            "networkProfile": {
                              "networkInterfaces": [
                                {
                                  "id": "[resourceId('Microsoft.Network/networkInterfaces', '/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/testrg/providers/Microsoft.Network/networkInterfaces/fakenic')]"
                                }
                              ]
                            }
                          }
                      }
                    ]
                  }
                }
              }
            ]
          }
        }
    }
  ]
}