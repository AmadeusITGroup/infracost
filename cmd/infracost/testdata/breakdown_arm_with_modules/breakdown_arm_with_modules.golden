Project: infracost/infracost/cmd/infracost/testdata/breakdown_arm_with_modules/breakdown_arm_with_modules.json

 Name                                                       Monthly Qty  Unit                      Monthly Cost   
                                                                                                                  
 Microsoft.Compute/virtualMachines/Linux/basic_a2                                                                 
 ├─ Instance usage (Linux, pay as you go, Basic_A2)                 730  hours                           $57.67   
 └─ os_disk                                                                                                       
    ├─ Storage (S4, LRS)                                              1  months                           $1.54   
    └─ Disk operations                               Monthly cost depends on usage: $0.0005 per 10k operations    
                                                                                                                  
 Microsoft.Compute/virtualMachines/Linux/vm_basic                                                                 
 ├─ Instance usage (Linux, pay as you go, Basic_A2)                 730  hours                           $57.67   
 └─ os_disk                                                                                                       
    ├─ Storage (S4, LRS)                                              1  months                           $1.54   
    └─ Disk operations                               Monthly cost depends on usage: $0.0005 per 10k operations    
                                                                                                                  
 OVERALL TOTAL                                                                                         $118.41 

*Usage costs can be estimated by updating Infracost Cloud settings, see docs for other options.

──────────────────────────────────
2 cloud resources were detected:
∙ 2 were estimated

┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━┳━━━━━━━━━━━━┓
┃ Project                                                          ┃ Baseline cost ┃ Usage cost* ┃ Total cost ┃
┣━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━━━╋━━━━━━━━━━━━━╋━━━━━━━━━━━━┫
┃ infracost/infracost/cmd/infraco...reakdown_arm_with_modules.json ┃          $118 ┃           - ┃       $118 ┃
┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━━━┻━━━━━━━━━━━━━┻━━━━━━━━━━━━┛

Err:

